test/e2e/apigatewayv2/tests/test_api.py

```diff
@@ -0,0 +1,539 @@
+import time
+
+import boto3
+import pytest
+import requests
+import random
+import string
+
+from apigatewayv2 import SERVICE_NAME, service_marker
+from apigatewayv2.bootstrap_resources import get_bootstrap_resources
+from apigatewayv2.replacement_values import REPLACEMENT_VALUES
+import apigatewayv2.tests.helper as helper
+from apigatewayv2.tests.helper import ApiGatewayValidator
+from common import k8s
+from common.aws import get_aws_region, get_aws_account_id
+from common.resources import load_resource_file
+
+DELETE_WAIT_AFTER_SECONDS = 10
+UPDATE_WAIT_AFTER_SECONDS = 10
+DEPLOYMENT_WAIT_AFTER_SECONDS = 10
+
+apigw_validator = ApiGatewayValidator(boto3.client('apigatewayv2'))
+test_resource_values = REPLACEMENT_VALUES.copy()
+
+
+@pytest.fixture(scope="module")
+def api_resource():
+    api_resource_name = test_resource_values['API_NAME']
+    api_ref, api_data = helper.api_ref_and_data(api_resource_name=api_resource_name,
+                                                replacement_values=test_resource_values)
+    if k8s.get_resource_exists(api_ref):
+        pytest.fail(f"expected {api_resource_name} to not exist. Did previous test cleanup?", False)
+    logging.debug(f"http api resource. name: {api_resource_name}, data: {api_data}")
+
+    k8s.create_custom_resource(api_ref, api_data)
+    cr = k8s.wait_resource_consumed_by_controller(api_ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(api_ref)
+
+    api_id = cr['status']['apiID']
+    test_resource_values['API_ID'] = api_id
+
+    yield api_ref, cr
+
+    k8s.delete_custom_resource(api_ref)
+    time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+
+@pytest.fixture(scope="module")
+def integration_resource(api_resource):
+    integration_resource_name = test_resource_values['INTEGRATION_NAME']
+    integration_ref, integration_data = helper.integration_ref_and_data(
+        integration_resource_name=integration_resource_name,
+        replacement_values=test_resource_values)
+    if k8s.get_resource_exists(integration_ref):
+        pytest.fail(f"expected {integration_resource_name} to not exist. Did previous test cleanup?", False)
+    logging.debug(f"apigatewayv2 integration resource. name: {integration_resource_name}, data: {integration_data}")
+
+    k8s.create_custom_resource(integration_ref, integration_data)
+    cr = k8s.wait_resource_consumed_by_controller(integration_ref)
+
+    assert cr is not None
+    assert k8s.get_resource_exists(integration_ref)
+
+    integration_id = cr['status']['integrationID']
+    test_resource_values['INTEGRATION_ID'] = integration_id
+
+    yield integration_ref, cr
+
+    k8s.delete_custom_resource(integration_ref)
+    time.sleep(DELETE_WAIT_AFTER_SECONDS)
+
+
+@pytest.fixture(scope="module")
+def authorizer_resource(api_resource):
+    authorizer_resource_name = test_resource_values['AUTHORIZER_NAME']
+    authorizer_uri = "arn:aws:apigateway:{region}:lambda:path/2015-03-31/functions/{authorizer_function_arn}/invocations" \
```

Comment:
nit: f-strings are the preferred method of string formatting

Reply:
Done.